### PR TITLE
Update to ember-cli 2.18

### PR DESCRIPTION
**Pull request by Gaurav0**
Notes:

1. The addon DOES NOT WORK in Ember 3.5+ due to breaking changes, so I allowed ember-beta to fail for now. (Ember 3.5+ requires id and until in all calls to deprecate())
2. The addon still (ironically) throws deprecation warnings in ember-cli 2.18+ due to use of read/rebuild broccoli apis.

Passes all tests except in scenarios ember-beta and ember-canary locally.

Builds on @rwjblue 's PR from here: https://github.com/mixonic/ember-cli-deprecation-workflow/pull/47

Addressed comments in that PR.

### CODE REVIEW

**Comment by rwjblue** [comment]
> The addon DOES NOT WORK in Ember 3.5+ due to breaking changes,

What breaking changes?

File: tests/unit/deprecation-collector-test.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ window.deprecationWorkflow.config = {
 };`);
 });
 
+// Fails in Ember 3.5+
 test('calling flushDeprecations without ID returns full message plus an additional deprecation for missing ID', (assert) => {
```
Comment (rwjblue) [inline tests/unit/deprecation-collector-test.js]: Lets guard these tests in Ember >= 3.

Specifically:

```js
import hasEmberVersion from '@ember/test-helpers/has-ember-version';

if (!hasEmberVersion(3,0)) {
  // tests here 
}
```

Comment (Gaurav0) [reply tests/unit/deprecation-collector-test.js]: done, tests guarded.

**Comment by Gaurav0** [comment]
Can we turn on travis again?

**Comment by rwjblue** [comment]
Hmm, I didn’t think we disabled it 🤔

@mixonic - can you toggle Travis on this repo?

**Comment by Gaurav0** [comment]
@rwjblue can you remove changes requested?

**Comment by rwjblue** [comment]
Thanks for working on this! I also pinged @mixonic in discord, but he likely hasn't had time to get CI sorted.

**Pull request merged**